src/auth/Readme.md

```diff
@@ -0,0 +1,266 @@
+```
+
+## How does auth handle resetting password?
+
+#### **`SaleorAuthClient.resetPassword: ({ email, password, token }: PasswordResetVariables) => Promise<PasswordResetResponse>`**
+
+SaleorAuthClient class provides you with a reset password method. If the reset password mutation is successful, it'll log you in automatically just like after a regular sign in. The [onSignIn method of useAuthChange hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout) will also be triggered.
+
+```javascript
+const { resetPassword } = useSaleorAuthContext();
+
+const response = await resetPassword({
+  email: "example@mail.com",
+  password: "newPassword",
+  token: "apiToken",
+});
+```
+
+## How do I use this with Urql?
+
+Once an authentication change happens, you might want to refetch some of your queries. Because Urql doesn't provide a direct way to invalidate cache manually, we're following urql's [proposed approach](https://github.com/urql-graphql/urql/issues/297#issuecomment-501646761) of installing a new instance of the client in place of the old one. We have a hook for that called `useUrqlClient` that takes Urql `ClientOptions` as an only argument and returns the current `urqlClient` and `resetClient` function:
+
+```javascript
+const { urqlClient, resetClient } = useUrqlClient({
+  url: saleorApiUrl,
+  fetch: saleorAuthClient.fetchWithAuth,
+  // other client props
+});
+```
+
+Then, you'll want to pass the `resetClient` function to the useAuthChange hook
+
+```javascript
+useAuthChange({
+  storage,
+  onSignedOut: () => resetClient(),
+  onSignedIn: () => resetClient(),
+});
+```
+
+### How to put it all together?
+
+```jsx
+export const App = () => {
+  const { saleorApiUrl } = getQueryParams();
+  const { locale, messages } = useLocale();
+  const saleorAuthClientProps = useSaleorAuthClient({
+    saleorApiUrl,
+    storage: localStorage,
+  });
+
+  const { saleorAuthClient } = saleorAuthClientProps;
+
+  const { urqlClient, resetClient } = useUrqlClient({
+    suspense: true,
+    requestPolicy: "cache-first",
+    url: saleorApiUrl,
+    fetch: saleorAuthClient.fetchWithAuth,
+  });
+
+  useAuthChange({
+    onSignedOut: () => resetClient(),
+    onSignedIn: () => resetClient(),
+  });
+
+  return (
+    <SaleorAuthProvider {...saleorAuthClientProps}>
+      <UrqlProvider value={urqlClient}>
+        <Home />
+      </UrqlProvider>
+    </SaleorAuthProvider>
+  );
+};
+```
+
+### How do I use it with Apollo?
+
+We have a hook for that called useApolloClient that authenticated fetch as its only argument and returns the current client and resetClient function
+
+```javascript
+const { apolloClient, resetClient } = useApolloClient(saleorAuthClient.fetchWithAuth);
+```
+
+Becasue we're using the client to also retrieve unauthenticated data in SSR, we separated them into two - one for SSR, that can be used outside of React flow, and the other returned by our hook.
```

Comment:
```suggestion
Because we're using the client to also retrieve unauthenticated data in SSR, we separated them into two - one for SSR, which can be used outside of React flow, and the other returned by our hook.
```